coast/config/config_parser.py

```diff
@@ -0,0 +1,99 @@
+"""Config parser."""
+import json
+from pathlib import Path
+from typing import Union
+
+from .config_structure import (
+    ConfigTypes,
+    ConfigKeys,
+    GriddedConfig,
+    IndexedConfig,
+    Dataset,
+    Domain,
+    CodeProcessing
+)
+
+class ConfigParser():
+    """Class for parsing gridded and indexed configuration files."""
+    def __init__(self, json_path: Union[Path, str]):
+        """Config parser constructor.
+        
+        Args:
+            json_path (Union[Path, str]): path to json config file.
+        """
+        with open(json_path, 'r') as j:
+            json_content = json.loads(j.read())
```

Comment:
I would suggest using a dataclass to hold the config values rather than a dict. For one thing it makes accurate type hinting easier.

Reply:
Again I think for an MVP, the dict is ok. Going down the full dataclass and typing can be added to the black as future work